oqsprov/oqsprov_keys.c

```diff
@@ -619,9 +619,10 @@ OQSX_KEY *oqsx_key_new(OSSL_LIB_CTX *libctx, char* oqs_name, char* tls_name, int
 #ifdef USE_ENCODING_LIB
         if (alg_idx >= 0 && oqs_alg_encoding_list[alg_idx] != NULL) {
             if (qsc_encoding_by_name_oid(&ret->oqsx_encoding_ctx.encoding_ctx, &ret->oqsx_encoding_ctx.encoding_impl, oqs_oid_alg_list[2*alg_idx], oqs_alg_encoding_list[alg_idx]) != QSC_ENC_OK) {
-                fprintf(stderr, "Could not create OQS signature encoding algorithm %s (%s, %s). Defaulting to no encoding.\n", oqs_alg_encoding_list[alg_idx], oqs_name, oqs_oid_alg_list[2*alg_idx]);
-            ret->oqsx_encoding_ctx.encoding_ctx = NULL;
+                fprintf(stderr, "Could not create OQS signature encoding algorithm %s (%s, %s).\n", oqs_alg_encoding_list[alg_idx], oqs_name, oqs_oid_alg_list[2*alg_idx]);
```

Comment:
OK -- you don't output that you're not defaulting to no encoding. But the code doesn't change -- so still defaults on error, correct?

Reply:
it adds a `goto err`, which will raise an error.

Reply:
Oops -- didn't see that. Thanks.